From fcc319526d28ad4a4b2833ef940ce2445dd60b8b Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Fri, 19 Jul 2019 16:29:14 +0200
Subject: [PATCH 359/386] mmc: mmc_oops: set proper timeout for data command

The mmc_oops feature can be used with different card types each having its
own timeout data characteristic. Make usage of mmc_set_data_timeout in
order to set proper timeout value and then use it while waiting for oops
request completion.

Change-Id: I646784eaaf18fcc8ce3ec5a2f267fee9893c7968
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/12816
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Marcin Wojtas <marcin@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/mmc/core/mmc_oops.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mmc/core/mmc_oops.c b/drivers/mmc/core/mmc_oops.c
index 82bb207235c8..ceed30a61b84 100644
--- a/drivers/mmc/core/mmc_oops.c
+++ b/drivers/mmc/core/mmc_oops.c
@@ -93,6 +93,7 @@ static void mmc_panic_write(struct mmcoops_context *cxt,
 	data.sg = &sg;
 	data.sg_len = 1;
 
+	mmc_set_data_timeout(&data, card);
 	mmc_wait_for_oops_req(host, &mrq);
 
 	if (cmd.error)
-- 
2.17.1

